### ISSUE TITLE
installsAfter looks for an exact match even when version is missing.

### ISSUE THREAD

**Issue by brendandburns** [devcontainers/cli#345]
I have read through https://github.com/devcontainers/cli/issues/163 and though it is related, I think this is a different issue.

It appears that the `installAfter` feature is looking for an exact string match rather than a prefix.

I want to be able to say:
```json
    "installsAfter": [
        "ghcr.io/devcontainers/features/common-utils"
    ]
```

and have that match:
```json
   "features": {
       "ghcr.io/devcontainers/features/common-utils:1.0.0"
   }
```

I suspect the problem is here:
https://github.com/devcontainers/cli/blob/d4fbd73b30cedd50c185b90b4ae51200481f1000/src/spec-configuration/containerFeaturesOrder.ts#L57

Where there is a `Map.get(featureId)` call which won't match the feature with the semantic version above.

**Comment by samruddhikhandale** [comment]
Hi 👋 

Thanks for reporting. I already have a PR out which fixes this bug - https://github.com/devcontainers/cli/pull/335/files#r1055939808

**Comment by samruddhikhandale** [comment]
Released the fix with v0.27.0 🚀
Feel free to re-open this issue if the bug resurfaces, thanks!
